src/torchmetrics/image/fid.py

```diff
@@ -205,22 +168,15 @@ def _compute_fid(mu1: Tensor, sigma1: Tensor, mu2: Tensor, sigma2: Tensor, eps:
         sigma1: covariance matrix over activations calculated on predicted (x) samples
         mu2: mean of activations calculated on target (y) samples
         sigma2: covariance matrix over activations calculated on target (y) samples
-        eps: offset constant - used if sigma_1 @ sigma_2 matrix is singular
 
     Returns:
         Scalar value of the distance between sets.
     """
-    diff = mu1 - mu2
-
-    covmean = sqrtm(sigma1.mm(sigma2))
-    # Product might be almost singular
-    if not torch.isfinite(covmean).all():
-        rank_zero_info(f"FID calculation produces singular product; adding {eps} to diagonal of covariance estimates")
-        offset = torch.eye(sigma1.size(0), device=mu1.device, dtype=mu1.dtype) * eps
-        covmean = sqrtm((sigma1 + offset).mm(sigma2 + offset))
+    a = (mu1 - mu2).square().sum(dim=-1)
+    b = sigma1.trace() + sigma2.trace()
+    c = torch.linalg.eigvals(sigma1 @ sigma2).sqrt().real.sum(dim=-1)
 
-    tr_covmean = torch.trace(covmean)
-    return diff.dot(diff) + torch.trace(sigma1) + torch.trace(sigma2) - 2 * tr_covmean
+    return a + b - 2 * c
 
 
 class FrechetInceptionDistance(Metric):
```

Comment:
The formula should be

```tex
FID = \|\mu - \mu_w\|^2 + tr(\Sigma + \Sigma_w - 2(\Sigma \Sigma_w)^{\frac{1}{2}})
```

and the note about `scipy` is not necessary anymore.

Reply:
Nice catch. Could you add it here as a suggestion? :)

Reply:
> Could you add it here as a suggestion?

I couldn't write a suggestion because that part of the code hadn't been changed.

> I have taken of this now :]

Thx!